gcp/modules/gpii-istio/network_policy.tf

```diff
@@ -5,9 +5,57 @@ resource "kubernetes_network_policy" "deny-default" {
   }
 
   spec {
-    pod_selector = {}
+    pod_selector {}
+
+    policy_types = ["Ingress", "Egress"]
+
+    egress {
+      # Allow egress traffic to Istio control plane
+      to {
+        namespace_selector {
+          match_labels = {
+            k8s-app = "istio"
+          }
+        }
+      }
+
+      # Allow egress traffic within namespace
+      to {
+        namespace_selector {
+          match_labels = {
+            name = "gpii"
+          }
+        }
+      }
+
+      # Allow egress traffic to dns
+      to {
+        namespace_selector {
+          match_labels = {
+            name = "kube-system"
+          }
+        }
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+    }
+
+    egress {
+      # Allow egress traffic Compute Metadata
+      to {
+        ip_block {
+          cidr = "10.16.0.0/20"
+        }
+      }
+      ports {
+        port     = "8181"
+        protocol = "TCP"
+      }
+    }
 
```

Comment:
What about traffic to GCP APIs (metrics, trace)? Also traffic to flowmanager public endpoint needs to be allowed for SSL check to work, in case that PR merges first.